docs/source/codedoc.rst

```diff
@@ -318,3 +318,45 @@ The content of ``my_project/__init__.py`` includes::
     from .core._impl import MyClass
 
     __all__ = ("MyClass",)
+
+Branch priorities
+-----------------
+
+When pydoctor deals with try/except/else or if/else block, it makes sure that the names defined in 
+the "principal" branch do not get overriden by names defined in the except hanlders or ifs' else block. 
```

Comment:
This strikes me as a bit awkward.  I think you want to be saying something about the priority of the binding of the name in the *last* branch? Or the `else` branch?

Reply:
No it's not about whether the branch if last or not. It's only the  except hanlders and ifs' else blocks where we enable the override name guard.

Reply:
the `else` branch of `try:` blocks have the priority over the `except:` block. Similarly, the main body of `if:` blocks have priority over what's defined in the `else:` block. 

So the thing that might be confusing here if that `Try.orelse` and `If.orelse` are NOT handled with the same override priority. Because the main branches in case of a `if:` is the `body` uniquely and in case of a `try:` it's the `body`, the  `orelse` and the `finalbody`, in this order. So just saying `else:` block can be confusing.